src/community/backup-restore/core/src/main/java/org/geoserver/backuprestore/writer/StyleInfoAdditionalResourceWriter.java

```diff
@@ -0,0 +1,40 @@
+/* (c) 2014 Open Source Geospatial Foundation - all rights reserved
+ * (c) 2001 - 2016 OpenPlans
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.backuprestore.writer;
+
+import java.io.IOException;
+
+import org.geoserver.backuprestore.Backup;
+import org.geoserver.backuprestore.utils.BackupUtils;
+import org.geoserver.catalog.StyleInfo;
+import org.geoserver.platform.resource.Resource;
+import org.geoserver.platform.resource.Resources;
+
+/**
+ * TODO
+ * 
+ * @author Alessio Fabiani, GeoSolutions
+ *
+ */
+public class StyleInfoAdditionalResourceWriter
+        implements CatalogAdditionalResourcesWriter<StyleInfo> {
+
+    @Override
+    public boolean canHandle(Object item) {
+        return (item instanceof StyleInfo);
+    }
+
+    @Override
+    public void writeAdditionalResources(Backup backupFacade, Resource base, StyleInfo item)
+            throws IOException {
+        Resource sldFile = backupFacade.getGeoServerDataDirectory().get(item, item.getFilename());
```

Comment:
Not necessarily a SLD file, could be a css or ysld one, better call it styleFile